app/controllers/concerns/application_notifications_concern.rb

```diff
@@ -0,0 +1,14 @@
+module ApplicationNotificationsConcern
+  extend ActiveSupport::Concern
+
+  # Returns the view path of the notification
+  #
+  # @param [Course::Notification|UserNotification] notification The target notification
+  # @return [String] The view path of the notification
+  def notification_view_path(notification)
+    activity = notification.activity
+    root_path = "#{activity.notifier_type}/#{activity.event}"
+    notification_class_name = notification.class.name.underscore.sub('/', '_').pluralize
```

Comment:
`gsub`. what happens when you're using `Course::Something::SomethingElse::Notification`? that's why I asked so many times, `sub` or `gsub`. @allenwq fixed this in another gem recently. I would imagine he should be more sensitive to it.